app/src/test/java/com/kickstarter/viewmodels/NativeCheckoutRewardViewHolderViewModelTest.kt

```diff
@@ -569,54 +569,71 @@ class NativeCheckoutRewardViewHolderViewModelTest : KSRobolectricTestCase() {
         val reward = project.backing()?.reward() as Reward
         setUpEnvironment(environment())
 
-        this.vm.inputs.projectAndReward(project, reward)
+        this.vm.inputs.configureWith(ProjectDataFactory.project(project), reward)
 
         this.vm.inputs.rewardClicked(3)
         this.showPledgeFragment.assertNoValues()
         this.koalaTest.assertNoValues()
     }
 
     @Test
-    fun testShowPledgeFragment_WhenProjectIsLive() {
+    fun testShowPledgeFragment_WhenProjectIsLiveNotBacked() {
         val reward = RewardFactory.reward()
         val liveProject = ProjectFactory.project()
         setUpEnvironment(environment())
 
-        this.vm.inputs.projectAndReward(liveProject, reward)
+        this.vm.inputs.configureWith(ProjectDataFactory.project(liveProject), reward)
         this.showPledgeFragment.assertNoValues()
 
         // When a reward from a live project is clicked, start checkout.
         this.vm.inputs.rewardClicked(2)
         this.showPledgeFragment.assertValue(Pair.create(liveProject, reward))
         this.koalaTest.assertValue("Select Reward Button Clicked")
+        this.lakeTest.assertValue("Select Reward Button Clicked")
+    }
+
+    @Test
+    fun testShowPledgeFragment_WhenProjectIsLiveBacked() {
+        val reward = RewardFactory.reward()
+        val backedProject = ProjectFactory.backedProject()
+        setUpEnvironment(environment())
+
+        this.vm.inputs.configureWith(ProjectDataFactory.project(backedProject), reward)
+        this.showPledgeFragment.assertNoValues()
+
+        // When a reward from a live backed project is clicked, start checkout.
+        this.vm.inputs.rewardClicked(2)
+        this.showPledgeFragment.assertValue(Pair.create(backedProject, reward))
+        this.koalaTest.assertValue("Select Reward Button Clicked")
+        this.lakeTest.assertNoValues()
```

Comment:
I'm a little confused by this test 🤔 shouldn't the lake client also fire "Select Reward Button Clicked"?

Reply:
no, i'm only firing for unbacked projects; this test is clicking a reward of a backed project (changing your reward)